design_docs/support_external_swift_packages.md

```diff
@@ -0,0 +1,204 @@
+# Proposal for Representing External Swift Packages
+
+## Problem
+
+External Swift packages are, typically, represented in Bazel projects in one of two ways:
+
+1. As an external repositry using `http_archive` providing a custom build file.
+2. As an external repository using `spm_repositories` and `spm_pkg` from
+   [cgrindel/rules_spm](https://github.com/cgrindel/rules_spm/). 
+
+If an Xcode project is generated to build with Bazel (BWB), no additional support is required in the
+Xcode project for the project to build properly. Whatever is defined in the Bazel project will
+handle the building and linking of the external dependencies.
+
+If an Xcode project is generated to build with Xcode (BWX), the Xcode project needs to understand
+how to incorporate the external dependency.
+
+## Possible Solutions
+
+### Solution 1: Treat the External Dependency as Another Target
+
+In essence, the external dependencies are imported as Xcode targets. 
+
+The pros for this approach are:
+
+- The generation code can leverage much of its existing logic mapping Bazel targets.
+- This maps well for Bazel projects that download the external package using `http_archive` and
+  define a custom build file.
+
+The cons for this approach are:
+
+- The generated project will be cluttered with targets that are not relevant to the developer.
+  These will include not just the direct dependencies, but all of the transitive dependencies for
+  the external packages.
```

Comment:
I assume we might have a different way to filter out targets that aren't relevant to a particular developer for their project in the future, so this would be a problem only until we have a solution for it that could be applicable for this as well.

Reply:
Perhaps. If the files are used for compilation, they need to exist in the project somewhere. Is there a way in Xcode to hide targets, but still use them for compilation.

Reply:
Yes, but it involves a more sophisticated Bazel Generated Files step (or additional similar targets). I'll detail something like that soon, as it's needed for other use cases anyway.

Reply:
But also in Build with Bazel, we won't need most files to be part of the project.